client/src/templates/Challenges/classic/DesktopLayout.js

```diff
@@ -27,6 +27,18 @@ const reflexProps = {
 };
 
 class DesktopLayout extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { showNotes: false, showPreview: true, showConsole: false };
+    this.switchDisplayTab = this.switchDisplayTab.bind(this);
+  }
+
+  switchDisplayTab(displayTab) {
+    this.setState({
+      [displayTab]: !this.state[displayTab]
+    });
```

Comment:
```suggestion
    this.setState(state => {
      return {
        [displayTab]: !state[displayTab]
      };
```
To avoid bugs when multiple state changes get batched.